src/views/LogTab.vue

```diff
@@ -90,14 +144,71 @@ export default defineComponent({
     IonIcon
   },
   setup() {
+
+    const router = useRouter();
+
+    interface Item {
+      name: string;
+      calories: number;
+    }
+
+    interface UndoItems {
+      name: string;
+      calories: number;
+      index: number;
+    }
+
+    const removeItemsState = ref(false);
     const timeStamp = new Date();
+
+    function removeItem(item: Item) {
+      const index = items.value.indexOf(item);
+      undoStack.value.push({ 
+        name: item.name, 
+        calories: item.calories, 
+        index 
+      });
+      items.value.splice(index, 1);
+    }
+
+    const undoStack = ref<UndoItems[]>([]);
```

Comment:
Also why is this defined after it's used?

Reply:
changed that